conans/client/build/cmake.py

```diff
@@ -307,9 +307,17 @@ def build(self):
                     # current package
                     path_content = tools.load(path)
                     for dep in self._conanfile.deps_cpp_info.deps:
-                        from_str = self._conanfile.deps_cpp_info[dep].rootpath
+                        from_str = self._format_path_to_cmake_config(self._conanfile.deps_cpp_info[dep].rootpath)
                         # try to replace only if from str is found
                         if path_content.find(from_str) != -1:
                             dep_str = "${CONAN_%s_ROOT}" % dep.upper()
                             self._conanfile.output.info("Patching paths for %s: %s to %s" % (dep, from_str, dep_str))
                             tools.replace_in_file(path, from_str, dep_str, strict=False)
+
+    def _format_path_to_cmake_config(self, pathstr):
```

Comment:
I think we could move (tab) this function into the `patch_config_paths` to limit its scope.

Reply:
ok, will do that.